src/task/apply_rule.rs

```diff
@@ -43,12 +39,14 @@ impl ApplyRuleTask {
         }
 
         for plan in new_plans {
-            let group_plan = optimizer_ctx.memo_mut().copy_in_plan(Some(group.clone()), &plan);
+            let group_plan = optimizer_ctx
+                .memo_mut()
+                .copy_in_plan(Some(self.plan.borrow().group().clone()), &plan);
             if group_plan.borrow().operator().is_logical() {
                 task_runner.push_task(OptimizePlanTask::new(group_plan));
             } else {
-                let required_prop: PhysicalProperties = PhysicalProperties::new();
-                let new_task = EnforceAndCostTask::new(group_plan, Rc::new(required_prop.clone()));
+                let required_prop = optimizer_ctx.required_properties();
```

Comment:
`optimizer_ctx.required_properties()` is the top required properties for root group, here we should use the required properties for current group.